check_patroni/types.py

```diff
@@ -35,44 +36,37 @@ class PatroniResource(nagiosplugin.Resource):
     def rest_api(self: "PatroniResource", service: str) -> Any:
         """Try to connect to all the provided endpoints for the requested service"""
         for endpoint in self.conn_info.endpoints:
-            try:
-                cert: Optional[Union[Tuple[str, str], str]] = None
-                verify: Optional[Union[str, bool]] = None
-                if endpoint[:5] == "https":
-                    if self.conn_info.cert is not None:
-                        # we can have: a key + a cert or a single file with key and cert.
-                        cert = self.conn_info.cert
-                    if self.conn_info.ca_cert is not None:
-                        # if cert is not None: this is the CA certificate
-                        # otherwise this is a ca bundle with root certificate
-                        # then some optional intermediate certificate and finally
-                        # the cerver certificate to validate the certification chain
-                        verify = self.conn_info.ca_cert
-                    else:
-                        if cert is None:
-                            # if cert is None we want to bypass https verification,
-                            # this is in secure and should be avoided for production use
-                            verify = False
-
-                _log.debug(
-                    f"Trying to connect to {endpoint}/{service} with cert: {cert} verify: {verify}"
-                )
+            cert: Optional[Union[Tuple[str, str], str]] = None
+            verify: Optional[Union[str, bool]] = None
+            if urlparse(endpoint).scheme == "https":
+                if self.conn_info.cert is not None:
+                    # we can have: a key + a cert or a single file with key and cert.
+                    cert = self.conn_info.cert
+                if self.conn_info.ca_cert is not None:
+                    # if cert is not None: this is the CA certificate
+                    # otherwise this is a ca bundle with root certificate
+                    # then some optional intermediate certificate and finally
+                    # the server certificate to validate the certification chain
+                    verify = self.conn_info.ca_cert
+
+            _log.debug(
+                f"Trying to connect to {endpoint}/{service} with cert: {cert} verify: {verify}"
+            )
```

Comment:
Another thing I forgot to mention: log messages should not be formatted directly and this the job of the logging formatter. So instead you should write this as:
```suggestion
            _log.debug(
                "Trying to connect to %(endpoint)s/%(service)s with cert: %(cert)s verify: %(verify)s",
                {"endpoint": endpoint, "service": service, "cert": cert, "verify": verify},
            )
```

The reason is that formatting thus happens at runtime *only* if the message is to be emitted, i.e. if the level is at least `DEBUG` here. Otherwise, it's done unconditionally, which is a waste if the level is higher than `DEBUG` here. Sure, a simple string formatting is usually cheap, but sometimes its not (it in fact depends on the value type to be formatted), and if it repeated a lot, it might matter.